### ISSUE TITLE
BuildTestCommand should not use shell=True

### ISSUE THREAD

**Issue by vsoch** [buildtesters/buildtest#182]
Currently, `buildtest.tools.system.BuildTestCommand` issues a command to the host with shell=True

```
        self.ret = subprocess.Popen(
            cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
        )
```
From what I've read, this [should be avoided](https://stackoverflow.com/questions/3172470/actual-meaning-of-shell-true-in-subprocess) - we can likely compensate for expanding envars / other features of having the shell through other measures.

**Comment by shahzebsiddiqui** [comment]
Yea i know ``shell=True`` is a security vulnerability and yes if there is an alternative to running shell commands i would be for it. The code would need to be refactored as i recall ``BuildTestCommand`` is used in several places.

**Comment by vsoch** [comment]
Note: we will also need to update the command in spider.py to use this. Comment here https://github.com/HPC-buildtest/buildtest-framework/pull/181#pullrequestreview-364382563

**Comment by vsoch** [comment]
I'd like to work on this after the class refactor for the types, which is a larger chunk of work.

**Comment by shahzebsiddiqui** [comment]
Sounds good with me.

**Comment by vsoch** [comment]
This is fixed in #219
